### DISCUSSION TITLE
How to use the Narrator on Text-generation-webui?

### DISCUSSION THREAD

**Discussion by Ex3cutionerX** [erew123/alltalk_tts#174] [Q&A]
I'm trying to use the narrator on text-generation-webui, but it doesn't seem to work. I've already enabled it and tried putting the text in asterisks * text * or quotation marks "text", but it still doesn't work when I click preview audio button. I even reinstalled text-gen-webui and the Alltalktts. Everything else seems to function fine, like Deepspeed. Am I doing something wrong? Can someone help me?

**Comment by erew123** [comment]
Hi @Ex3cutionerX 

The preview doesn't go through the narration code, so you wont get that effect.

The Narration is aimed at the AI speaking to you. So if you use a character within Text-gen-webui it should work fine (the kind you can find in the Gallery or any other character you can add). https://github.com/oobabooga/text-generation-webui/wiki/Custom-characters-FAQ/86cbfc13644f26571bb484fbfce8ba5955dc0d0f

Is there some other effect you are trying to get? 

Thanks

**Reply by Ex3cutionerX** [reply]
Thank you, I've downloaded a model and the narration works perfectly when I talk to the AI. I just want to know now what the button 'Permanently replace audio with the message texts' means?

**Reply by erew123** [reply]
Ah. Something I forgot to change in the text. It means do or don't show "text+audio" or just the audio. It's the setting for the "show text" box in the text gen interface.

**Reply by Ex3cutionerX** [reply]
And how can i make my own written dialogue to be read by the narrator * and the character " since the preview don't work with the narration on text-gen-webui? If i instruct the AI from my downloaded models to repeat my text, it never reads it correctly, sometimes it remove the asterisks and the end result is read wrong by the character part and the narrator. How can i do it with my own text?

**Reply by erew123** [reply]
Hi @Ex3cutionerX 

You can make calls from a terminal/command prompt via the API https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-example-command-lines-standard-generation

Beyond that, please wait until I have v2 released as you will have a full interface to do as you wish:

![image](https://github.com/erew123/alltalk_tts/assets/35898566/98cb7723-f7c1-4d58-90e3-fbff9ee7c645)

Thanks

**Reply by Ex3cutionerX** [reply]
I've tried to use it via CMD with curl, and the result are great only in English language. If i try to put it in language=pt (portuguese), when i use special characters from that language, it reads it wrong, the characters ê or é results in Ãª Ã© on the API request, what can i do? Thanks.
